### DISCUSSION TITLE
Is it possible to set custom max width?

### DISCUSSION THREAD

**Discussion by Demianeen** [neovim/neovim#28221] [Q&A]
I usually have a nvim on full screen, and when I open Markdown, my line rarely wraps because of this. Is it possible to set the max width before wrap to 100 for example instead of the window length?

**Comment by thallada** [comment]
I'm baffled why this has never been implemented in vim in the past 34 years. It is such a basic function that literally any other markdown editor in existence has. Every plugin that claims to do soft wrapping only does it at the edge of the window, not at any arbitrary width no matter the window width.

Is this some fundamental limitation of the architecture of vim/neovim? I would think not since you can resize splits easily enough and that soft wraps text fine.

I know someone will probably say that I should hard wrap lines instead, but I don't want to do that because that inserts newline characters into the file. When editing markdown, I only want to insert newlines at paragraph breaks. The line length is a display-only property that should not affect the raw text.

If neovim could implement this I could finally do all note-taking in neovim instead of various other apps like obsidian.

**Reply by Aitanuqui** [reply]
@thallada, does

> Maybe you are looking set columns, or does it mess things up somehow 🤔

from @mclang work for you?

**Reply by thallada** [reply]
`set columns=80` is close to what I want. But the issue is that it affects the entire neovim window. So the status bar and tab bar and everything is limited to 80 characters even though I have thousands of horizontal pixels unused. And, if I open up a vertical split, then the 80 chars is divided into two 40 char splits making it even smaller, so I can't do vertical splits with that setting on.

**Comment by mclang** [comment]
Maybe you are looking `set columns`, or does it mess things up somehow :thinking:

**Comment by superatomic** [comment]
Might get added in https://github.com/neovim/neovim/pull/38167 or https://github.com/vim/vim/pull/20754, but currently that isn't a feature. Until then, your best bet is probably to just create an empty split window to the right and resize it accordingly.

Edit: Oh oops this is a two year old discussion; it just got bumped up from recent activity. Didn't notice.
